src/TreeHouse/Cache/Driver/FileDriver.php

```diff
@@ -58,13 +56,24 @@ public function get($key)
      */
     public function set($key, $value, $ttl = 0)
     {
-        $file = $this->getCacheFile($key);
+        $file = $this->getCacheFilePath($key);
+
+        $this->ensureDirectory(pathinfo($file, PATHINFO_DIRNAME));
+
+        // write to tmp file first
+        $tmpFile = tempnam($file, 'swap');
+        chmod($tmpFile, 0664);
 
-        if (false === file_put_contents($file, $value)) {
+        if (false === file_put_contents($tmpFile, $value)) {
             throw new \RuntimeException(sprintf('Could not write to cache file: "%s"', $file));
         }
 
-        return true;
+        // now move the tmp file to ensure atomic writes
```

Comment:
Don't think you ensure atomic writes with this. 
If a big value gets written, then quickly a really small one (which will finish earlier). Then the old big key overwrites the new small key which is the real value.

What about flock? http://php.net/manual/en/function.flock.php